src/erl_optics_server.erl

```diff
@@ -0,0 +1,118 @@
+-module(erl_optics_server).
+-include("erl_optics.hrl").
+
+-behaviour(gen_server).
+
+-export([init/1,
+         handle_call/3,
+         handle_cast/2,
+         handle_info/2,
+         terminate/2,
+         code_change/3]).
+
+-export([start_link/1,
+         stop/0]).
+
+%for prototyping only
+-export([start/0,
+         test_update/0,
+         start_test/1]).
+
+-define(SERVER, ?MODULE).
+
+-record(state, {port = ?ENV_PORT     :: inet:port_number(),
```

Comment:
Are you sure you want to set the default values of these to `port` and `hostname`? I think it might be more clear if they are set to `undefined` by default.